clients/privacy-center/features/consent/types.ts

```diff
@@ -0,0 +1,23 @@
+export type ConsentItem = {
+  fidesDataUseKey: string;
+  name: string;
+  description: string;
+  highlight: boolean;
+  url: string;
+  defaultValue: boolean;
+  consentValue?: boolean;
+};
+
+export type ApiUserConsent = {
+  data_use: string;
+  data_use_description?: string;
+  opt_in: boolean;
+};
+
+export type ApiUserConsents = {
+  consent?: ApiUserConsent[];
+};
+
+export type DataUseConsent = {
+  [dataUse: string]: boolean | undefined;
+};
```

Comment:
Open Question: Is the format of `datause -> boolean` what we want?
- This is different from the [pro cookie](https://support.ethyca.com/docs/how-to-configure-consent-for-ccpas-do-not-sell-my-personal-information#understanding-ethycas-global-consent-object--cookie) which is concerned with particular platforms and generic "groups".
- However, my guess of the use case is that this flexibility of taxonomy and consent is a good thing.
- If we don't want data uses as the keys, are we hard-coding certain taxonomy rules?
- Note that because the keys are fides keys, the dots can be a little confusing. The object has to be accessed as `duc['provide.service']`. Trying to do `duc.provide.service` wouldn't work.

Reply:
Answer: Make it even more flexible by letting the user define what the assigned keys are!